src/pages/index.js

```diff
@@ -22,96 +34,233 @@ const profileDescriptionInput = profileEditModal.querySelector(
+    .then((cardData) => {
+      renderCard(cardData);
+      newCardPopup.close();
+    })
+    .catch((err) => console.log(err))
+    .finally(() => {
+      submitButton.textContent = initialButtonText;
+    });
+});
 newCardPopup.setEventListeners();
 
-//Preview Image Popup
+// Preview Image Popup
 const previewImagePopup = new PopupWithImage("#preview-modal");
 previewImagePopup.setEventListeners();
 
-//Section
+// Section to render cards
 const section = new Section(
   {
-    items: initialCards,
+    items: [],
     renderer: renderCard,
   },
   ".cards__list"
 );
-section.renderItems();
 
-//UserInfo
-const user = new UserInfo(".profile__title", ".profile__description");
+api
+  .getInitialCards()
+  .then((cards) => {
+    section.renderItems(cards);
+  })
+  .catch((err) => console.log("Error fetching cards:", err));
+
+// Edit Avatar Popup
+const avatarEditPopup = new PopupWithForm(
+  "#profile-avatar-modal",
+  (formData) => {
+    const submitButton = avatarEditPopup._popupElement.querySelector(
+      "#modal-profile-submit"
+    );
+    const initialButtonText = submitButton.textContent;
+    submitButton.textContent = "Saving...";
+
+    api
+      .updateAvatar(formData.link)
+      .then((res) => {
+        user.changeAvatar(res.avatar);
+        avatarEditPopup.close();
+      })
+      .catch((err) => console.log("Error updating avatar:", err))
+      .finally(() => {
+        submitButton.textContent = initialButtonText;
+      });
+  }
+);
+avatarEditPopup.setEventListeners();
 
 /* -------------------------------------------------------------------------- */
 /*                                  Functions                                 */
 /* -------------------------------------------------------------------------- */
-function handleImageClick(name, link) {
-  previewImagePopup.open(name, link);
+
+// Render card function
+function renderCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDelete,
+    handleLikeCard
+  );
+  const cardElement = card.getView();
+  section.addItem(cardElement);
 }
 
-function handleProfileEditSubmit(profileData) {
-  const name = profileData.title;
-  const description = profileData.subheader;
-  user.setUserInfo(name, description);
-  editProfilePopup.close();
+// Handle image click
+function handleImageClick(card) {
+  previewImagePopup.open({ name: card.name, link: card.link });
 }
 
-function handleAddCardSubmit(newCardData, cardListElement) {
-  const name = newCardData.title;
-  const alt = newCardData.title;
-  const link = newCardData.url;
-  renderCard({ name, alt, link });
-  newCardPopup.close();
+// Handle card delete
+function handleDelete(card) {
+  confirmation.open();
+  confirmation.setConfirmSubmit(() => {
+    api
+      .deleteCard(card._id)
+      .then(() => {
+        confirmation.close();
+        card._handleDeleteCard();
+      })
+      .catch((err) => console.log(err));
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors